### PR TITLE
fix: position tooltip within chart with single value xScale

### LINKED ISSUE #254: Single datum chart on element hover tooltip overflows the chart
**Describe the bug**
When we only have a single datum chart, the tooltip that appears on hover is positioned either to the left or the right of the element, pushing it outside of the chart.

**To Reproduce**
1. create single datum chart
2. hover over chart element
3. see tooltip overflowing chart

**Expected behavior**
Tooltip should be visible within the chart.  For single datum charts, we probably want the tooltip to follow the cursor within the element.

**Screenshots**
single_datum_tooltip

**Version (please complete the following information):**
 - OS: all
 - Browser all
 - Elastic Charts: 7.0.1

**Additional context**

**Errors in browser console**
none

**Kibana Cross Issues**
n/a

**Checklist**
- [x] every related Kibana issue is listed under `Kibana Cross Issues` list
- [x] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present

### PR DESCRIPTION
**Pull request by emmacunningham**
## Summary

fix #254 

This PR introduces a fix for the tooltip positioning on a chart with a single value xScale.  

I'm not entirely sure where the best place to position the tooltip, so this presents the minimal effort version, where the tooltip will be positioned inside of the chart aligned along the chart's edge (left if horizontal rotation, top if vertical rotation):

single_value

Implementation-wise, we've added a method `isSingleValue` to the `Scale` classes so that we can check if a scale is a single value scale (so that we can apply the different positioning in that case).  The `customDomain` knob was added to the stories to demonstrate that, when the chart only has one datapoint but has a multiple value scale, the tooltips are positioned as usual (since there is room to display them as normal).

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~- [ ] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)~
~- [ ] Proper documentation or storybook story was added for features that require explanation or tutorials~
- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [x] Unit tests were updated or added to match the most common scenarios
- [x] Each commit follows the [convention](https://github.com/elastic/elastic-charts/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by markov00** [comment]
hey @emmacunningham just a suggestion, what if in the "single value" mode we switch between the "crosshair/vertical" mode and we manually enable the "follow" mode?

File: stories/bar_chart.tsx [hunk 1]
```diff
@@ -855,9 +855,42 @@ storiesOf('Bar Chart', module)
...
+      {
+        '0 deg': 0,
+        '90 deg': 90,
+        '-90 deg': -90,
+        '180 deg': 180,
+      },
+      0,
+    );
+
+    const theme = mergeWithDefaultTheme(
```
Comment (nickofthyme) [inline stories/bar_chart.tsx]: No need to use `mergeWithDefaultTheme` anymore. Will merge with `LIGHT_THEME` by default.

File: stories/bar_chart.tsx [hunk 2]
```diff
@@ -872,11 +905,63 @@ storiesOf('Bar Chart', module)
...
```
Comment (nickofthyme) [inline stories/bar_chart.tsx]: Same here

File: stories/bar_chart.tsx (hunk 1 above)
Comment (emmacunningham) [reply stories/bar_chart.tsx]: https://github.com/elastic/elastic-charts/pull/259/commits/c2c13bd9e1e2aa97daa11f6eade317e4d80325ce

File: stories/bar_chart.tsx (hunk 2 above)
Comment (emmacunningham) [reply stories/bar_chart.tsx]: https://github.com/elastic/elastic-charts/pull/259/commits/c2c13bd9e1e2aa97daa11f6eade317e4d80325ce

**Comment by emmacunningham** [comment]
> hey @emmacunningham just a suggestion, what if in the "single value" mode we switch between the "crosshair/vertical" mode and we manually enable the "follow" mode?

@markov00 yes, good idea.  have implemented that logic here: https://github.com/elastic/elastic-charts/pull/259/commits/121a8fb8cf96bb633f3fdb0d041044462de5f475

**Pull request merged**